### PR TITLE
feat: improve import file to calendar and extend capabilities

### PR DESCRIPTION
**Pull request by Nir-P**
This is an extension for [import to calendar](https://github.com/PythonFreeCourse/calendar/pull/119):

### what's new?

- Support "Location" field when importing txt, csv or ics files 
  files must be in the following structure or the old structure (without Location)
  ```
  Title1, Content1, S_Date1, E_Date1, Location1
  Title2, Content2, S_Date2, E_Date2, Location2
  ...
  TitleN, ContentN, S_DateN, E_DateN, LocationN
  ```
- Start and End dates support hours and minutes format

**the new rules we set (can be changed through a global variable):**
-  The location field length is up to 50 characters.
-  The event duration limit.

**Additional tests added:**
- Start date is smaller or equal to the end date.
- The Location field may be empty.
- Checking that new files from the old and new format are working flawlessly.
- And some more here and there :)

### CODE REVIEW

File: app/internal/import_file.py [hunk 1]
```diff
@@ -45,26 +57,54 @@ def is_file_exist(file: str) -> bool:
...
     """
     check if date is valid and in the range according to the rule we have set
     """
     now_year = datetime.datetime.now().year
     if isinstance(date, str):
-        try:
-            check_date = datetime.datetime.strptime(date, DATE_FORMAT)
-        except ValueError:
+        check_date = change_string_to_date(date)
+        if check_date is False:
```
Comment (yammesicka) [inline app/internal/import_file.py:79]: Prefer `== False`

File: app/internal/import_file.py [hunk 2]
```diff
@@ -45,26 +57,54 @@ def is_file_exist(file: str) -> bool:
     return Path(file).is_file()
 
 
+def change_string_to_date(date: str) -> Union[datetime.datetime, bool]:
+    try:
+        if re.search(r":", date):
```
Comment (yammesicka) [inline app/internal/import_file.py:62]: Prefer `: in date`

File: app/internal/import_file.py [hunk 3]
```diff
@@ -45,26 +57,54 @@ def is_file_exist(file: str) -> bool:
...
+            date1 = datetime.datetime.strptime(date, DATE_FORMAT2)
+        else:
+            date1 = datetime.datetime.strptime(date, DATE_FORMAT)
+    except ValueError:
```
Comment (yammesicka) [inline app/internal/import_file.py:66]: What is the purpose of the except here?

**Review by yammesicka — CHANGES_REQUESTED** [review]
Great job! Added few of my insights :)

File: app/internal/import_file.py (hunk 3 above)
Comment (Nir-P) [reply app/internal/import_file.py:66]: if the string is not a valid date `datetime.strptime` will fail to transfer the string to date and we want to catch it.

File: app/internal/import_file.py (hunk 1 above)
Comment (Nir-P) [reply app/internal/import_file.py]: changed it to `not` if this is ok.

File: app/internal/import_file.py (hunk 2 above)
Comment (Nir-P) [reply app/internal/import_file.py]: Done

**Review by yammesicka — APPROVED** [review]

**Pull request merged**